### PR TITLE
(CDAP-16243) Changes in CDAP CRD and operator to mount system app configs

### PR DESCRIPTION
**Pull request by pandyajay10**
Adding support to add system config apps for CDAP. Changes include adding new map field CRD for system config apps with their json config. Also adding mechanism to create one file per system app via Volume mount configuration.

TESTS: tested by manual testing to verify correctly generated system app. Added unit test.

### CODE REVIEW

File: api/v1alpha1/cdapmaster_types.go [hunk 1]
```diff
@@ -68,6 +68,10 @@ type CDAPMasterSpec struct {
 	// 2:        UI in its own Pod. Other services in a single multi-container Pod
 	// 3:        UI and Router in their own Pod. All other services runs in a multi-container Pod
 	NumPods *int32 `json:"numPods,omitempty"`
+	// SystemAppConfigs specifies configs used by CDAP to run system apps
+	// dynamically. Each entry is of format <filename, json app config> which will
+	// create a separate system config file with entry value as file content.
+	SystemAppConfigs map[string]string `json:"systemappconfigs,omitempty"`
```
Comment (wyzhang) [inline api/v1alpha1/cdapmaster_types.go]: can you move this section up next to "Config"

File: controllers/cdapmaster_controller.go [hunk 2]
```diff
@@ -175,6 +175,13 @@ func (h *ConfigMapHandler) Objects(rsrc interface{}, rsrclabels map[string]strin
 		obj := buildConfigMapObject(spec)
 		expected = append(expected, obj)
 	}
+
+	// Creates system app config object. Creates one data object per system app config file.
+	sysAppConfigSpec := newConfigMapSpec(m, getObjName(m, configMapSysAppConf), mergeMaps(m.Labels, rsrclabels))
```
Comment (wyzhang) [inline controllers/cdapmaster_controller.go]: would you please refactor mergeMaps to a local var?

File: controllers/deployment.go [hunk 3]
```diff
@@ -227,7 +228,7 @@ func buildStatefulSets(master *v1alpha1.CDAPMaster, name string, services Servic
 }
 
 // Return a single single-/multi- container deployment containing a list of supplied services
-func buildDeployment(master *v1alpha1.CDAPMaster, name string, services ServiceGroup, labels map[string]string, cconf, hconf, dataDir string) (*DeploymentSpec, error) {
+func buildDeployment(master *v1alpha1.CDAPMaster, name string, services ServiceGroup, labels map[string]string, cconf, hconf, sysappconf, dataDir string) (*DeploymentSpec, error) {
```
Comment (wyzhang) [inline controllers/deployment.go:231]: The list of params is getting too long here. Let's refactor by creating a config class that encapsulates cconf, hconf and sysconf and pass that a single param. Either make the refactoring in this change or a follow-up one.

Comment (pandyajay10) [reply controllers/deployment.go:231]: Will take care of it in future refactoring.

File: config/crd/bases/cdap.cdap.io_cdapmasters.yaml [hunk 4]
```diff
@@ -203,7 +203,6 @@ spec:
             logLevels:
               additionalProperties:
                 type: string
-              description: LogLevels is a set of logger name to log level settings.
```
Comment (wyzhang) [inline config/crd/bases/cdap.cdap.io_cdapmasters.yaml]: Is this intentionally?

File: config/crd/bases/cdap.cdap.io_cdapmasters.yaml [hunk 5]
```diff
@@ -1154,6 +1153,15 @@ spec:
               description: ServiceAccountName is the service account for all the service
                 pods.
               type: string
+            systemappconfigs:
+              additionalProperties:
+                type: string
+              description: LogLevels is a set of logger name to log level settings.
+                SystemAppConfigs specifies configs used by CDAP to run system apps
```
Comment (wyzhang) [inline config/crd/bases/cdap.cdap.io_cdapmasters.yaml]: hm..can you check why LogLevels ... got placed here?

**Review by wyzhang — APPROVED** [review]

**Pull request merged**